### PR TITLE
avro: Initial integration

### PR DESCRIPTION
**Pull request by henryrneh**
Integrating Apache Avro into OSS-Fuzz.

### CODE REVIEW

File: projects/avro/Dockerfile
Comment (martin-g) [inline projects/avro/Dockerfile:19]: Why 3.8.1 ?
There are several newer versions. Current latest is 3.9.2.

File: projects/avro/build.sh
Comment (martin-g) [inline projects/avro/build.sh:23]: Java 15 ?! It is not maintained since long time.
Most projects use the Long Time Supported ones - Java 11 or 17. Others use the latest - 20.

File: projects/avro/project-parent/fuzz-targets/pom.xml
Comment (martin-g) [inline projects/avro/project-parent/fuzz-targets/pom.xml:13]: See above

File: projects/avro/project-parent/fuzz-targets/pom.xml
Comment (martin-g) [inline projects/avro/project-parent/fuzz-targets/pom.xml:59]: Better use 3.x, e.g. 3.1.0. It has better support for JUnit 5.x

**Review by Alan32Liu — APPROVED** [review]
Happy to merge after you address the comments above :)

File: projects/avro/Dockerfile
Comment (henryrneh) [reply projects/avro/Dockerfile]: Thank you for the feedback, already updated to 3.9.2.

File: projects/avro/build.sh
Comment (henryrneh) [reply projects/avro/build.sh]: In OSS-Fuzz the default Java version is 15, we will update it to newer LTS Java version in the future, thank you for your feed back, it is changed to 11 now.

**Comment by henryrneh** [comment]
> Happy to merge after you address the comments above :)

Updated based on comments!

**Comment by henryrneh** [comment]
@martin-g Hello Martin, if you wish to be the maintainer of Avro projects on OSS-Fuzz feel free to give me your email, I will add you to project.yaml. You will receive emails regarding findings and other issues. If you do not wish to be maintainer at this moment feel free to contact me or OSS-Fuzz team in the future to add you.

**Comment by martin-g** [comment]
Hi @henryrneh !

Thank you for your work on this!
Please add `mgrigorov@apache.org`.
I am not very familiar with the Java Avro SDK, I maintain the Rust Avro SDK, but I will try to help however I can!

**Comment by martin-g** [comment]
Is there anything to be done from my side on https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=59743 ?
The logs suggest that there is a problem with the tooling itself.

**Comment by henryrneh** [comment]
Hi @martin-g thanks for reaching out, this is not an Avro issue and we will take care of this.

**Pull request merged**
